lib/content-services/src/lib/search/components/search-chip-input/search-chip-input.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatChip } from '@angular/material/chips';
+import { By } from '@angular/platform-browser';
+import { TranslateModule } from '@ngx-translate/core';
+import { Subject } from 'rxjs';
+import { ContentTestingModule } from '../../../testing/content.testing.module';
+import { SearchChipInputComponent } from './search-chip-input.component';
+
+describe('SearchChipInputComponent', () => {
+    let component: SearchChipInputComponent;
+    let fixture: ComponentFixture<SearchChipInputComponent>;
+    const onResetSubject = new Subject<void>();
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [SearchChipInputComponent],
+            imports: [
+                TranslateModule.forRoot(),
+                ContentTestingModule
+            ]
+        });
+
+        fixture = TestBed.createComponent(SearchChipInputComponent);
+        component = fixture.componentInstance;
+        component.onReset = onResetSubject.asObservable();
+        fixture.detectChanges();
+    });
+
+    afterEach(() => removeAllChips());
+
+    function getChipInput(): HTMLInputElement {
+        return fixture.debugElement.query(By.css('input')).nativeElement;
+    }
+
+    function getChipList(): MatChip[] {
+        return fixture.debugElement.queryAll(By.css('mat-chip')).map((chip) => chip.nativeElement);
+    }
+
+    function getChipValue(index: number): string {
+        return fixture.debugElement.queryAll(By.css('mat-chip span')).map((chip) => chip.nativeElement)[index].innerText;
+    }
+
+    function enterNewChip(value: string) {
+        const input = getChipInput();
+        input.value = value;
+        fixture.detectChanges();
+        input.dispatchEvent(new KeyboardEvent('keydown', {keyCode: 13}));
```

Comment:
can you use something different, keyCode seems deprecated? 
<img width="1097" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/a8619ad1-b875-4e2b-a4fb-d57fecfcbd32">

Reply:
I tried using `key` or `code` but none of them fired the event for me, using `keyCode` was the only way it worked for me, do you know any working alternative that can be used here?

Reply:
Thans for checking @MichalKinas. I see. I think it may be related with matChips implementation because normally it should work, see for example here: 
https://stackoverflow.com/questions/41135875/unable-to-simulate-keypress-event-in-angular-2-unit-test-jasmine

Then leave it :)